sync/device.ts

```diff
@@ -158,11 +158,19 @@ async function runSshCommand(address: string, command: string) {
 }
 
 export function stopSupervisor(address: string) {
-	return runSshCommand(address, 'systemctl stop resin-supervisor');
+	try {
+		return runSshCommand(address, 'systemctl stop balena-supervisor');
+	} catch (e) {
+		return runSshCommand(address, 'systemctl stop resin-supervisor');
```

Comment:
This catch block does not work as expected. Applying new supervisor to old OS failed at this step.

Reply:
This is probably because of the `return` statement, and can be fixed by replacing the `return`s with `await`.

Reply:
@cywang117 wouldn't I also need to change `stopSupervisor` to async for that to work?
```
'await' expressions are only allowed within async functions and at the top levels of modules.ts(1308)
```

Reply:
@klutchell Sorry, I didn't see the whole context. You can definitely change `stopSupervisor` to 
```
export async function stopSupervisor(address: string) {
	try {
		await runSshCommand(address, 'systemctl stop balena-supervisor');
	} catch {
		await runSshCommand(address, 'systemctl stop resin-supervisor');
}
```
the other alternative would be 
```
export function stopSupervisor(address: string) {
	return runSshCommand(address, 'systemctl stop balena-supervisor')
            .catch(() => runSshCommand(address, 'systemctl stop resin-supervisor'))
}
```
but `async/await` is more pretty :)

Reply:
Thank you @cywang117 !